MH4U Dex/MonsterDetailsViewController.m

```diff
@@ -29,43 +35,57 @@ @implementation MonsterDetailsViewController
 - (void)viewDidLoad
 {
     [super viewDidLoad];
-    [self.lowRankSubView setHidden:YES];
+    
+    [self segmentedControlChanged:self.segmentedControl];
 }
 
 - (IBAction)segmentedControlChanged:(UISegmentedControl *)segmentedControl
 {
+    self.overviewSubView.hidden = YES;
+    self.lowRankSubView.hidden = YES;
+    self.highRankSubView.hidden = YES;
+    self.GRankSubView.hidden = YES;
+    
     switch (segmentedControl.selectedSegmentIndex) {
         case Overview:
-            [self.overviewSubView setHidden:NO];
-            [self.lowRankSubView setHidden:YES];
-            self.navigationItem.title = [NSString stringWithFormat:@"%@ Overview", self.monster];
+            self.overviewSubView.hidden = NO;
+            self.title = [NSString stringWithFormat:@"%@ Overview", self.monster];
             break;
         case LowRank:
-            [self.overviewSubView setHidden:YES];
-            [self.lowRankSubView setHidden:NO];
-            self.navigationItem.title = [NSString stringWithFormat:@"%@ LR Drops", self.monster];
+            self.lowRankSubView.hidden = NO;
+            self.title = [NSString stringWithFormat:@"%@ LR Drops", self.monster];
+            break;
+        case HighRank:
+            self.highRankSubView.hidden = NO;
+            self.title = [NSString stringWithFormat:@"%@ HR Drops", self.monster];
+            break;
+        case GRank:
+            self.GRankSubView.hidden = NO;
+            self.title = [NSString stringWithFormat:@"%@ GR Drops", self.monster];
             break;
         default:
             break;
     }
 }
 
-- (void)viewWillAppear:(BOOL)animated
-{
-    [super viewWillAppear:animated];
-    NSLog(@"Test");
-}
-
 - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
 {
-    if ([segue.identifier isEqualToString:@"embedLowRank"]) {
-        MonsterLowRankTableViewController *lowRankVC = (MonsterLowRankTableViewController *)segue.destinationViewController;
-        lowRankVC.managedObjectContext = self.managedObjectContext;
-        lowRankVC.monster = self.monster;
-    } else if ([segue.identifier isEqualToString:@"embedOverview"]) {
+    
+    if ([segue.identifier isEqualToString:@"embedOverview"]) {
         MonsterOverviewViewController *overViewVC = (MonsterOverviewViewController *)segue.destinationViewController;
         overViewVC.managedObjectContext = self.managedObjectContext;
-        overViewVC.monster = self.monster;
+        overViewVC.monsterName = self.monster;
+    } else {
+        MonsterItemTableViewController *itemVC = (MonsterItemTableViewController *)segue.destinationViewController;
+        itemVC.managedObjectContext = self.managedObjectContext;
+        itemVC.monsterName = self.monster;
```

Comment:
Is this (and 4 lines farther up) another instance of a property named `monster` that's a name string rather than a `Monster` object?